EvtWeightTools/src/Data_to_MC_CorrectionInterface_0l_4tau_trigger.cc

```diff
@@ -40,132 +40,30 @@ Data_to_MC_CorrectionInterface_0l_4tau_trigger::getSF_triggerEff(TriggerSFsys ce
   {
-    const double prob_tau1_data = getProb_tau(tau1_status, eff_2tau_tauLeg1_data);
-    const double prob_tau1_mc   = getProb_tau(tau1_status, eff_2tau_tauLeg1_mc);
-
-    for(int tau2_status = k2tau; tau2_status <= kNot2tau; ++tau2_status)
-    {
-      const double prob_tau2_data = getProb_tau(tau2_status, eff_2tau_tauLeg2_data);
-      const double prob_tau2_mc   = getProb_tau(tau2_status, eff_2tau_tauLeg2_mc);
-
-      for(int tau3_status = k2tau; tau3_status <= kNot2tau; ++tau3_status)
-      {
-        const double prob_tau3_data = getProb_tau(tau3_status, eff_2tau_tauLeg3_data);
-        const double prob_tau3_mc   = getProb_tau(tau3_status, eff_2tau_tauLeg3_mc);
-
-        for(int tau4_status = k2tau; tau4_status <= kNot2tau; ++tau4_status)
-        {
-          const double prob_tau4_data = getProb_tau(tau4_status, eff_2tau_tauLeg4_data);
-          const double prob_tau4_mc   = getProb_tau(tau4_status, eff_2tau_tauLeg4_mc);
-
-          int nTrig_2tau_tauLeg = 0;
-          if(tau1_status == k2tau)
-          {
-            ++nTrig_2tau_tauLeg;
-          }
-          if(tau2_status == k2tau)
-          {
-            ++nTrig_2tau_tauLeg;
-          }
-          if(tau3_status == k2tau)
-          {
-            ++nTrig_2tau_tauLeg;
-          }
-          if(tau4_status == k2tau)
-          {
-            ++nTrig_2tau_tauLeg;
-          }
-
-          const bool isTrig_2tau_toy = nTrig_2tau_tauLeg >= 2;
-
-          if(isTriggered_2tau_ == isTrig_2tau_toy)
-          {
-            prob_data += prob_tau1_data * prob_tau2_data * prob_tau3_data * prob_tau4_data;
-            prob_mc   += prob_tau1_mc   * prob_tau2_mc   * prob_tau3_mc   * prob_tau4_mc;
-          }
-        }
-      }
-    }
+    prob_data *= tau_leg_efficiency(matched_hadTau->pt(), matched_hadTau->decayMode(), "ditau", wp_str_, "eff_data", sys);
+    prob_mc *= tau_leg_efficiency(matched_hadTau->pt(), matched_hadTau->decayMode(), "ditau", wp_str_, "eff_mc", sys);
   }
 
-  if(isDEBUG_)
-  {
-    std::cout << "hadTau (lead):    pT = " << hadTau1_pt_ << ", eta = " << hadTau1_eta_ << "\n"
-                 "hadTau (sublead): pT = " << hadTau2_pt_ << ", eta = " << hadTau2_eta_ << "\n"
-                 "hadTau (third):   pT = " << hadTau3_pt_ << ", eta = " << hadTau3_eta_ << "\n"
-                 "hadTau (fourth):  pT = " << hadTau4_pt_ << ", eta = " << hadTau4_eta_ << "\n"
-                 "eff (X_tau1): data = " << eff_2tau_tauLeg1_data   << ", MC = " << eff_2tau_tauLeg1_mc   << "\n"
-                 "eff (X_tau2): data = " << eff_2tau_tauLeg2_data   << ", MC = " << eff_2tau_tauLeg2_mc   << "\n"
-                 "eff (X_tau3): data = " << eff_2tau_tauLeg3_data   << ", MC = " << eff_2tau_tauLeg3_mc   << "\n"
-                 "eff (X_tau4): data = " << eff_2tau_tauLeg4_data   << ", MC = " << eff_2tau_tauLeg4_mc   << '\n'
-    ;
-  }
+  if ( isDEBUG_ )
+    std::cout << "prob_data: " << prob_data << "\t prob_mc: " << prob_mc << std::endl;
```

Comment:
If you don't want to use curly braces, then I recommend that you put them on a single line, otherwise please use curly braces like in other places of this codebase. Also, `std::endl` is computationally expensive because it forces to flush the stream. Consider using just the newline character (`'\n'`) instead.